mypy/build.py

```diff
@@ -705,8 +705,8 @@ def __init__(
         self.quickstart_state = read_quickstart_file(options, self.stdout)
         # Fine grained targets (module top levels and top level functions) processed by
         # the semantic analyzer, used only for testing. Currently used only by the new
-        # semantic analyzer.
-        self.processed_targets: list[str] = []
+        # semantic analyzer. Tuple of module and target name.
+        self.processed_targets: list[tuple[str, str]] = []
```

Comment:
`processed_targets` is (as the comment says) for testing only, specifically for `TypeCheckSuite`.

I'm adding the module's name so that `TypeCheckSuite` could perform filtering more coherently:
```diff
 actual = [
-    t
-    for t in actual
-    if not any(t.startswith(mod) for mod in core_modules + ["mypy_extensions"])
+    target
+    for module, target in res.manager.processed_targets
+    if module in testcase.test_modules
 ]
```